TF_DATA_DIR containing absolute path is parsed as relative path

## Description

If `TF_DATA_DIR` contains an absolute path, it's parsed as a relative path from `$TF_DATA_DIR/modules/modules.json`. The debug log shows the error quite clearly:

```shell
export TF_DATA_DIR=/builds/terraform-data

terraform init                       
Initializing modules...
Downloading terraform-aws-modules/vpc/aws 2.21.0 for vpc...
- vpc in /builds/terraform-data/modules/vpc/terraform-aws-modules-terraform-aws-vpc-2417f60

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "aws" (hashicorp/aws) 2.42.0...

Terraform has been successfully initialized!

❯ TFLINT_LOG=debug tflint --module --deep
[...]
08:27:10 loader.go:56: [INFO] Initialize new loader
08:27:10 terraform.go:55: [INFO] TF_DATA_DIR environment variable found: /builds/terraform-data
08:27:10 loader.go:67: [INFO] Module manifest file found. Initializing...
08:27:10 terraform.go:55: [INFO] TF_DATA_DIR environment variable found: /builds/terraform-data
08:27:10 loader.go:262: [DEBUG] Parsing the module manifest file: {"Modules":[{"Key":"","Source":"","Dir":"."},{"Key":"vpc","Source":"terraform-aws-modules/vpc/aws","Version":"2.21.0","Dir":"/builds/terraform-data/modules/vpc/terraform-aws-modules-terraform-aws-vpc-2417f60"}]}
08:27:10 loader.go:81: [INFO] Load configurations under .
08:27:10 loader.go:96: [INFO] Module inspection is enabled. Building a root module with children...
08:27:10 loader.go:244: [DEBUG] Trying to load the module: key=vpc, version=2.21.0, dir=builds/terraform-data/modules/vpc/terraform-aws-modules-terraform-aws-vpc-2417f60
08:27:10 loader.go:103: [ERROR] Failed to load modules: <nil>: Failed to read module directory; Module directory builds/terraform-data/modules/vpc/terraform-aws-modules-terraform-aws-vpc-2417f60 does not exist or cannot be read.
Failed to load configurations. 1 error(s) occurred:

Error: Failed to read module directory

Module directory builds/terraform-data/modules/vpc/terraform-aws-modules-terraform-aws-vpc-2417f60 does not exist or cannot be read.
```

If I'm not mistaken, we could check for an absolute path here:

https://github.com/terraform-linters/tflint/blob/000c2eb1a92bb43b1107372ea409bd7f2008caac/tflint/loader.go#L240-L243

If this is the right place, I'll file a PR.

## Versions

```shell
terraform version && tflint --version
Terraform v0.12.18
TFLint version 0.13.2
```

--- a/tflint/loader.go
+++ b/tflint/loader.go
@@ -238,6 +238,10 @@ func (l *Loader) moduleWalker() configs.ModuleWalker {
 		}
 
 		dir := filepath.Join(l.currentDir, record.Dir)
+		if filepath.IsAbs(record.Dir) {
+			// If record.Dir is an absolute path, leave it
+			dir = record.Dir
+		}
 		if record.Root != "" {
 			dir = filepath.Join(dir, record.Root)
 		}

